src/mon/MDSMonitor.cc

```diff
@@ -554,7 +554,7 @@ bool MDSMonitor::prepare_update(MonOpRequestRef op)
     } catch (const bad_cmd_get& e) {
       bufferlist bl;
       mon.reply_command(op, -EINVAL, e.what(), bl, get_last_committed());
-      return true;
+      return false; /* nothing to propose */
```

Comment:
The return value for prepare_update is true if the update has been *handled*. It has nothing to do with whether the paxos system needs to do a propose or not.

So this change is definitely bad.

Reply:
Cc @rishabh-d-dave as we discussed this verbally.

Glad the header docs are getting updated